exporter/azuremonitorexporter/trace_to_envelope.go

```diff
@@ -133,6 +133,7 @@ func spanToEnvelopes(
 		}
 
 		spanEventEnvelope := newEnvelope(span, toTime(spanEvent.Timestamp()).Format(time.RFC3339Nano))
+		spanEventEnvelope.Tags[contracts.OperationParentId] = traceutil.SpanIDToHexOrEmptyString(span.SpanID())
```

Comment:
`newEnvelope` method has the entry for `envelope.Tags[contracts.OperationParentId]`, that seems to be correct. 
`envelope.Tags[contracts.OperationParentId] = traceutil.SpanIDToHexOrEmptyString(span.ParentSpanID())`


> spanEventEnvelope.Tags[contracts.OperationParentId] = traceutil.SpanIDToHexOrEmptyString(span.SpanID())

Setting the current span ID as the parent does not seem correct. Did you validate with an app that issue #27233 is resolved by this update?

Providing the .NET implementation here for reference - https://github.com/Azure/azure-sdk-for-net/blob/2d1996d14d430c541c5beabad612b98a01fb4a4e/sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/Customizations/Models/TelemetryItem.cs#L19

Java Implementation - https://github.com/Azure/azure-sdk-for-java/blob/9e6bab9bb50d865ed8b4e5ab3198e3fc56d9e468/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java#L288

Reply:
To be honest I haven't validated the change in conjunction with AppInsights. However the result looks correct:
Before the change:
![grafik](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/969523/ee13c8a1-491d-44ce-b580-a31ec3abc842)
Now:
![grafik](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/969523/7fb553b4-eaf2-4d55-b40a-ed2e750e0ecc)

Reply:
@rajkumar-rangaraj - the nifty difference here - we are not mapping spans, we are mapping logs - e.g. compare with                 https://github.com/Azure/azure-sdk-for-java/blob/9e6bab9bb50d865ed8b4e5ab3198e3fc56d9e468/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/LogDataMapper.java#L186 and lines around.

Reply:
Yes, this is correct in the context of logs. All logs will have the parent ID labeled as 'spanid.' However, you are modifying the trace envelope, which corresponds to spans. Logs are handled differently; therefore, the change should be applied to the log exporter file. The current proposed change might break the UI in other scenarios.

Reply:
Moving your changes to `log_to_envelope.go` should resolve an issue.

Reply:
@rajkumar-rangaraj - first huge sorry - with the `LogDataMapper`I've pointed you to the wrong direction. We are still and only talking about trace data. However I've found the more correct line that underpins my change also in the `azure-sdk-for-java` - https://github.com/Azure/azure-sdk-for-java/blob/9e6bab9bb50d865ed8b4e5ab3198e3fc56d9e468/sdk/monitor/azure-monitor-opentelemetry-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java#L779 
For `Event`s the parent span is the span of the current operation. That also makes sense, because the events are just annotations of the current operation/tracing span. So maybe we should take one step back, and find out how far we already agree.
1. There is a problem with the original implementation (#27233) - the original behaviour could and should be improved?
2. The provided PR solves this issue - and has no unintended side effects?
3. Is the form of the changes okay, or should the patch be rephrased?

Reply:
Got it. It makes sense for span events.